src/Kestrel.Core/Internal/Http2/Http2Stream.cs

```diff
@@ -362,10 +362,18 @@ public override void Abort(ConnectionAbortedException abortReason)
             AbortCore(abortReason);
         }
 
+        protected override void ErrorAfterHeaders()
+        {
+            // We can no longer change the response, send a Reset instead.
+            base.ErrorAfterHeaders();
+            var abortReason = new ConnectionAbortedException(CoreStrings.Http2StreamErrorAfterHeaders);
+            ResetAndAbort(abortReason, Http2ErrorCode.INTERNAL_ERROR);
+        }
+
         protected override void ApplicationAbort()
         {
             var abortReason = new ConnectionAbortedException(CoreStrings.ConnectionAbortedByApplication);
-            ResetAndAbort(abortReason, Http2ErrorCode.CANCEL);
+            ResetAndAbort(abortReason, Http2ErrorCode.INTERNAL_ERROR);
```

Comment:
Why the change here?

Reply:
I was reviewing the description and decided Cancel wasn't appropriate for the server to send. Cancel  is intended for the request initiator (the client in this case).